Services/Object/classes/class.ilObjectGUI.php

```diff
@@ -783,6 +783,21 @@ protected function addDidacticTemplateOptions(array &$a_options): void
     {
     }
 
+    protected function addAdoptContentLinkToToolbar(): void
+    {
+        if ($this->object->getType() === 'crs' || $this->object->getType() === 'grp') {
```

Comment:
I think this line is a strong hint, that this does not belong to the object service at all, imo. You usually do not want to add these type specific checks to general classes like ilObjectGUI since violates SoC principles. Additionally I have no idea how a non-container would make use of adopt content at all, so this is a container feature.

Reply:
Thank you very much @alex40724 for the quick reply.
But: You did look at the code, right? This here is just a trace I forgot to remove, coming from the previous implementation in `ilObjContainerGUI`. I will remove the if-statement right away as it is not needed anymore.

Why I would like to have this in ilObjectGUI for the time being (this time is not going to be very long):
* 'Adopt Content' is part of the copy process and copying is a concern of the `Object` right now (and most probably rightly so).
* I would like to start refactoring `Object` and this makes it (see the previous point) just this little bit easier.

Thanks again and best,
@kergomard